libraries/standard/http/src/http_client.c

```diff
@@ -725,68 +801,138 @@ static HTTPStatus_t sendHttpHeaders( const HTTPTransportInterface_t * pTransport
 
 /*-----------------------------------------------------------*/
 
-static HTTPStatus_t sendHttpBody( const HTTPTransportInterface_t * pTransport,
-                                  const uint8_t * pRequestBodyBuf,
-                                  size_t reqBodyBufLen )
+static HTTPStatus_t sendContentLength( const HTTPTransportInterface_t * pTransport,
+                                       size_t contentLength )
 {
     HTTPStatus_t returnStatus = HTTP_SUCCESS;
-    int32_t transportStatus = 0;
+
+    /* The content-length is the final header line sent. It is sent in this
+     * function before the body to utilize the reqBodyBufLen. */
+    char pContentLengthBuffer[ HTTP_CONTENT_LENGTH_FIELD_LEN +
```

Comment:
The arithmetic can be stored as a constant for better readability
```
#define HTTP_CONTENT_LENGTH_HEADER_MAX_LEN                              \
    ( HTTP_CONTENT_LENGTH_FIELD_LEN + HTTP_HEADER_FIELD_SEPARATOR_LEN + \
      MAX_INT32_NO_OF_DECIMAL_DIGITS + HTTP_HEADER_END_INDICATOR_LEN )
```

Reply:
Sure